### DISCUSSION TITLE
Pan & Zoom feature on mermaid.js output (SVG)

### DISCUSSION THREAD

**Discussion by amenella** [outline/outline#6932] [Ideas]
Hi,

A Zoom & Pan feature on mermaid.js output (which is an SVG) would be very much appreciated (especially for large diagrams) :pray: 

Something similar to what is available on mermaid.js live editor would be perfect, for example ([link](https://mermaid.live/edit#pako:eNptkVFvgjAUhf9Kc5-2DA0gViB7WeZM9rAn3wzJcoUGG2jr2hKnzv--wgY6XZ_u_W7PyentEXJVMEghr9GYOcdSo8gkcedJcoE1efwajci8yatbuuBmc0tXbK3xD07JA5eWYMmu8dJqLktSMlkwfTlsJeYNhSvv7q8GAi3rYRe7i3f8AWQwXTOsnlWt9DAwOy56oWs_Gsyrvj9d-rUPG_xGbXbDD-xVLhizA85RvqD9V9-t4BxorVRNuHnf8boYoG7koAUPBNMCeeF-otNlYDdMsAxSVxaoqwwy2d7DxqrlXuaQWt0wD5pt4dbx-3E93KKE9AifkAYzOp5EYRhPZ0FAp5R6sIc0DsdhEiY0jiI_CWc0OnlwUMrp_XGcJJOQ-jH1o2gSBNPObdUNW_PTN11gpl8)):

![image](https://github.com/outline/outline/assets/9946612/38aff706-97a0-48a9-9c7c-676dc8e5efdf)

I'm not sure if this a feature already included by latest versions of mermaid.js (which seems complicated to integrate from what I read in https://github.com/outline/outline/discussions/6470).

Anyway, thanks a lot for outline! :100:

**Comment by amenella** [comment]
Github seems to have a similar feature already integrated, for example in https://github.com/outline/outline/issues/6674, it is possible to pan & zoom the SVG file

**Comment by tommoor** [comment]
What I would really like to do is have the lightbox component be updated so that it's much more fully featured, as part of this it can allow panning and zooming on _all_ image types and just include Mermaid diagrams as part of that. 

The plan is also to allow navigating through images/diagrams in a document from within the lightbox as well.

**Reply by chulucninh09** [reply]
I think I raised duplicated request on #8213.

Can we start to discuss how to implement that? I can help for PR

**Comment by AlexBsrd** [comment]
Hey,

Up on this thread because we also are struggling with complex diagram being unreadable (forcing us to copy paste the content inside mermaid.compl/play to zoom on it while presenting features... 😕)

Any updates on this issue?

Regards, Alex

**Comment by LouisACR** [comment]
First of all, our team loves Outline, we use it all, but we use a lot of mermaid diagrams, and it's quite frustrating to be able to zoom in and out of the diagram, especially when it gets quite big.

A simple zooming system could be a great addition to outline.

I could try and implement it with a PR if needed, please let us know :)
